### PR TITLE
add skeleton code

### PR DESCRIPTION
**Pull request by carlory**
Example:

```sh
(⎈ |kind-kind:kube-system)➜  karmada-operator git:(skeleton) go run cmd/operator/operator.go -v4
I1130 11:36:17.953874   93466 flags.go:64] FLAG: --add-dir-header="false"
I1130 11:36:17.953924   93466 flags.go:64] FLAG: --alsologtostderr="false"
I1130 11:36:17.953926   93466 flags.go:64] FLAG: --concurrent-karmada-syncs="5"
I1130 11:36:17.953928   93466 flags.go:64] FLAG: --controllers="[*]"
I1130 11:36:17.953931   93466 flags.go:64] FLAG: --help="false"
I1130 11:36:17.953933   93466 flags.go:64] FLAG: --kube-api-burst="100"
I1130 11:36:17.953934   93466 flags.go:64] FLAG: --kube-api-qps="50"
I1130 11:36:17.953936   93466 flags.go:64] FLAG: --kubeconfig=""
I1130 11:36:17.953937   93466 flags.go:64] FLAG: --leader-elect="true"
I1130 11:36:17.953939   93466 flags.go:64] FLAG: --leader-elect-lease-duration="15s"
I1130 11:36:17.953941   93466 flags.go:64] FLAG: --leader-elect-renew-deadline="10s"
I1130 11:36:17.953942   93466 flags.go:64] FLAG: --leader-elect-resource-lock="leases"
I1130 11:36:17.953944   93466 flags.go:64] FLAG: --leader-elect-resource-name="karmada-operator"
I1130 11:36:17.953946   93466 flags.go:64] FLAG: --leader-elect-resource-namespace="karmada-system"
I1130 11:36:17.953975   93466 flags.go:64] FLAG: --leader-elect-retry-period="2s"
I1130 11:36:17.953979   93466 flags.go:64] FLAG: --log-backtrace-at=":0"
I1130 11:36:17.953983   93466 flags.go:64] FLAG: --log-dir=""
I1130 11:36:17.953984   93466 flags.go:64] FLAG: --log-file=""
I1130 11:36:17.953986   93466 flags.go:64] FLAG: --log-file-max-size="1800"
I1130 11:36:17.953988   93466 flags.go:64] FLAG: --log-flush-frequency="5s"
I1130 11:36:17.953989   93466 flags.go:64] FLAG: --log-json-info-buffer-size="0"
I1130 11:36:17.953992   93466 flags.go:64] FLAG: --log-json-split-stream="false"
I1130 11:36:17.953993   93466 flags.go:64] FLAG: --logging-format="text"
I1130 11:36:17.953995   93466 flags.go:64] FLAG: --logtostderr="true"
I1130 11:36:17.953996   93466 flags.go:64] FLAG: --one-output="false"
I1130 11:36:17.953997   93466 flags.go:64] FLAG: --resync-period="0s"
I1130 11:36:17.953999   93466 flags.go:64] FLAG: --skip-headers="false"
I1130 11:36:17.954052   93466 flags.go:64] FLAG: --skip-log-headers="false"
I1130 11:36:17.954057   93466 flags.go:64] FLAG: --stderrthreshold="2"
I1130 11:36:17.954059   93466 flags.go:64] FLAG: --v="4"
I1130 11:36:17.954062   93466 flags.go:64] FLAG: --version="false"
I1130 11:36:17.954064   93466 flags.go:64] FLAG: --vmodule=""
I1130 11:36:17.954087   93466 operator.go:103] Version: v0.0.0-master+$Format:%H$
I1130 11:36:17.954095   93466 operator.go:105] "Golang settings" GOGC="" GOMAXPROCS="" GOTRACEBACK=""
I1130 11:36:17.978867   93466 operator.go:174] Starting up "karmada"
I1130 11:36:17.978913   93466 operator.go:185] Started up "karmada"
I1130 11:36:17.978994   93466 internal.go:366] "Starting server" kind="health probe" addr="[::]:8443"
I1130 11:36:17.979012   93466 internal.go:366] "Starting server" path="/metrics" kind="metrics" addr="[::]:8080"
I1130 11:36:17.979047   93466 shared_informer.go:285] caches populated
I1130 11:36:17.979099   93466 leaderelection.go:248] attempting to acquire leader lease karmada-system/karmada-operator...
I1130 11:36:17.981369   93466 leaderelection.go:352] lock is held by zhanxis-MBP.lan_4a53e0d2-acf5-467a-b026-dc593473ba5d and has not yet expired
I1130 11:36:17.981380   93466 leaderelection.go:253] failed to acquire lease karmada-system/karmada-operator
I1130 11:36:20.916603   93466 leaderelection.go:352] lock is held by zhanxis-MBP.lan_4a53e0d2-acf5-467a-b026-dc593473ba5d and has not yet expired
I1130 11:36:20.916625   93466 leaderelection.go:253] failed to acquire lease karmada-system/karmada-operator
I1130 11:36:22.937316   93466 leaderelection.go:352] lock is held by zhanxis-MBP.lan_4a53e0d2-acf5-467a-b026-dc593473ba5d and has not yet expired
I1130 11:36:22.937336   93466 leaderelection.go:253] failed to acquire lease karmada-system/karmada-operator
I1130 11:36:25.312581   93466 leaderelection.go:352] lock is held by zhanxis-MBP.lan_4a53e0d2-acf5-467a-b026-dc593473ba5d and has not yet expired
I1130 11:36:25.312605   93466 leaderelection.go:253] failed to acquire lease karmada-system/karmada-operator
I1130 11:36:27.939422   93466 leaderelection.go:352] lock is held by zhanxis-MBP.lan_4a53e0d2-acf5-467a-b026-dc593473ba5d and has not yet expired
I1130 11:36:27.939443   93466 leaderelection.go:253] failed to acquire lease karmada-system/karmada-operator
I1130 11:36:30.947673   93466 leaderelection.go:352] lock is held by zhanxis-MBP.lan_4a53e0d2-acf5-467a-b026-dc593473ba5d and has not yet expired
I1130 11:36:30.947708   93466 leaderelection.go:253] failed to acquire lease karmada-system/karmada-operator
I1130 11:36:33.085412   93466 leaderelection.go:258] successfully acquired lease karmada-system/karmada-operator
I1130 11:36:33.085485   93466 recorder.go:103] "events: zhanxis-MBP.lan_590c201c-ddca-41d7-b49e-81e6f5f03713 became leader" type="Normal" object={Kind:Lease Namespace:karmada-system Name:karmada-operator UID:2814df43-e133-461c-877b-a9ef22293d19 APIVersion:coordination.k8s.io/v1 ResourceVersion:502228 FieldPath:} reason="LeaderElection"
I1130 11:36:33.085658   93466 controller.go:185] "Starting EventSource" controller="karmada" controllerGroup="operator.karmada.io" controllerKind="Karmada" source="kind source: *v1alpha1.Karmada"
I1130 11:36:33.085684   93466 controller.go:193] "Starting Controller" controller="karmada" controllerGroup="operator.karmada.io" controllerKind="Karmada"
I1130 11:36:33.086225   93466 reflector.go:221] Starting reflector *v1alpha1.Karmada (0s) from sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:262
I1130 11:36:33.086239   93466 reflector.go:257] Listing and watching *v1alpha1.Karmada from sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:262
I1130 11:36:33.186927   93466 shared_informer.go:285] caches populated
I1130 11:36:33.187016   93466 shared_informer.go:285] caches populated
I1130 11:36:33.187068   93466 controller.go:227] "Starting workers" controller="karmada" controllerGroup="operator.karmada.io" controllerKind="Karmada" worker count=5
I1130 11:36:40.394839   93466 karmada_controller.go:36] "Started syncing karmada" karmada="karmada-system/karmada" startTime="2022-11-30 11:36:40.39481 +0800 CST m=+22.460318542"
I1130 11:36:40.401272   93466 karmada_controller.go:86] "Reconciling karmada" name="karmada"
I1130 11:36:40.401292   93466 karmada_controller.go:38] "Finished syncing karmada" karmada="karmada-system/karmada" duration="6.474959ms"
I1130 11:36:40.401322   93466 karmada_controller.go:36] "Started syncing karmada" karmada="karmada-system/karmada" startTime="2022-11-30 11:36:40.401315 +0800 CST m=+22.466823876"
I1130 11:36:40.401334   93466 karmada_controller.go:86] "Reconciling karmada" name="karmada"
I1130 11:36:40.401337   93466 karmada_controller.go:38] "Finished syncing karmada" karmada="karmada-system/karmada" duration="19.375µs"
I1130 11:36:50.553945   93466 karmada_controller.go:36] "Started syncing karmada" karmada="karmada-system/karmada" startTime="2022-11-30 11:36:50.553926 +0800 CST m=+32.619571709"
I1130 11:36:50.553984   93466 karmada_controller.go:94] "Deleting unable gc resources" karmada="karmada-system/karmada"
I1130 11:36:50.559509   93466 karmada_controller.go:38] "Finished syncing karmada" karmada="karmada-system/karmada" duration="5.574833ms"
I1130 11:36:50.559571   93466 karmada_controller.go:36] "Started syncing karmada" karmada="karmada-system/karmada" startTime="2022-11-30 11:36:50.559562 +0800 CST m=+32.625208042"
I1130 11:36:50.559602   93466 karmada_controller.go:45] "Karmada has been deleted" karmada="karmada-system/karmada"
I1130 11:36:50.559607   93466 karmada_controller.go:38] "Finished syncing karmada" karmada="karmada-system/karmada" duration="42.459µs"
```

### CODE REVIEW

**Comment by carlory** [comment]
Operator Command:

```
Usage:
  karmada-operator [flags]

Generic flags:

      --controllers strings
                A list of controllers to enable. '*' enables all on-by-default controllers, 'foo' enables the controller named
                'foo', '-foo' disables the controller named 'foo'.
                All controllers: karmada
                Disabled-by-default controllers:  (default [*])
      --kube-api-burst int32
                Burst to use while talking with kubernetes apiserver. (default 100)
      --kube-api-qps float32
                QPS to use while talking with kubernetes apiserver. (default 50)
      --leader-elect
                Start a leader election client and gain leadership before executing the main loop. Enable this when running
                replicated components for high availability. (default true)
      --leader-elect-lease-duration duration
                The duration that non-leader candidates will wait after observing a leadership renewal until attempting to acquire
                leadership of a led but unrenewed leader slot. This is effectively the maximum duration that a leader can be
                stopped before it is replaced by another candidate. This is only applicable if leader election is enabled.
                (default 15s)
      --leader-elect-renew-deadline duration
                The interval between attempts by the acting master to renew a leadership slot before it stops leading. This must
                be less than or equal to the lease duration. This is only applicable if leader election is enabled. (default 10s)
      --leader-elect-resource-lock string
                The type of resource object that is used for locking during leader election. Supported options are 'leases',
                'endpointsleases' and 'configmapsleases'. (default "leases")
      --leader-elect-resource-name string
                The name of resource object that is used for locking during leader election. (default "karmada-operator")
      --leader-elect-resource-namespace string
                The namespace of resource object that is used for locking during leader election. (default "karmada-system")
      --leader-elect-retry-period duration
                The duration the clients should wait between attempting acquisition and renewal of a leadership. This is only
                applicable if leader election is enabled. (default 2s)
      --resync-period duration
                ResyncPeriod determines the minimum frequency at which watched resources are reconciled.

Karmada controller flags:

      --concurrent-karmada-syncs int
                The number of karmada controller reconcilers that are allowed to sync concurrently. (default 5)

Logs flags:

      --log-flush-frequency duration
                Maximum number of seconds between log flushes (default 5s)
      --log-json-info-buffer-size quantity
                [Alpha] In JSON format with split output streams, the info messages can be buffered for a while to increase
                performance. The default value of zero bytes disables buffering. The size can be specified as number of bytes
                (512), multiples of 1000 (1K), multiples of 1024 (2Ki), or powers of those (3M, 4G, 5Mi, 6Gi). Enable the
                LoggingAlphaOptions feature gate to use this.
      --log-json-split-stream
                [Alpha] In JSON format, write error messages to stderr and info messages to stdout. The default is to write a
                single stream to stdout. Enable the LoggingAlphaOptions feature gate to use this.
      --logging-format string
                Sets the log format. Permitted formats: "json" (gated by LoggingBetaOptions), "text".
                Non-default formats don't honor these flags: --add-dir-header, --alsologtostderr, --log-backtrace-at, --log-dir,
                --log-file, --log-file-max-size, --logtostderr, --one-output, --skip-headers, --skip-log-headers,
                --stderrthreshold, --vmodule.
                Non-default choices are currently alpha and subject to change without warning. (default "text")
  -v, --v Level
                number for the log level verbosity
      --vmodule pattern=N,...
                comma-separated list of pattern=N settings for file-filtered logging (only works for text log format)

Misc flags:

      --kubeconfig string
                Paths to a kubeconfig. Only required if out-of-cluster.

Global flags:

      --add-dir-header
                If true, adds the file directory to the header of the log messages (DEPRECATED: will be removed in a future
                release, see
                https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --alsologtostderr
                log to standard error as well as files (no effect when -logtostderr=true) (DEPRECATED: will be removed in a future
                release, see
                https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
  -h, --help
                help for karmada-operator
      --log-backtrace-at traceLocation
                when logging hits line file:N, emit a stack trace (default :0) (DEPRECATED: will be removed in a future release,
                see
                https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --log-dir string
                If non-empty, write log files in this directory (no effect when -logtostderr=true) (DEPRECATED: will be removed in
                a future release, see
                https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --log-file string
                If non-empty, use this log file (no effect when -logtostderr=true) (DEPRECATED: will be removed in a future
                release, see
                https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --log-file-max-size uint
                Defines the maximum size a log file can grow to (no effect when -logtostderr=true). Unit is megabytes. If the
                value is 0, the maximum file size is unlimited. (default 1800) (DEPRECATED: will be removed in a future release,
                see
                https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --logtostderr
                log to standard error instead of files (default true) (DEPRECATED: will be removed in a future release, see
                https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --one-output
                If true, only write logs to their native severity level (vs also writing to each lower severity level; no effect
                when -logtostderr=true) (DEPRECATED: will be removed in a future release, see
                https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --skip-headers
                If true, avoid header prefixes in the log messages (DEPRECATED: will be removed in a future release, see
                https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --skip-log-headers
                If true, avoid headers when opening log files (no effect when -logtostderr=true) (DEPRECATED: will be removed in a
                future release, see
                https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --stderrthreshold severity
                logs at or above this threshold go to stderr when writing to files and stderr (no effect when -logtostderr=true or
                -alsologtostderr=false) (default 2) (DEPRECATED: will be removed in a future release, see
                https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components)
      --version version[=true]
                Print version information and quit
```

**Review by lonelyCZ — COMMENTED** [review]
Reviewing is continue, and I think we can make the skeleton code same as `cmd/controller-manager` and `cmd/webhook` as far as possible. Expecting that there are some common function extracted in the future.

File: cmd/operator/app/operator.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	// never reach here
+	return nil
+}
+
+// InitFunc is used to launch a particular controller. It returns a controller
+// that can optionally implement other interfaces so that the controller manager
+// can support the requested features.
+// Any error returned will cause the controller process to `Fatal`
+// The bool indicates whether the controller was enabled.
+type InitFunc func(ctx context.Context, controllerCtx ControllerContext) (enabled bool, err error)
```
[the remark this answers is not in the document]
Comment (carlory) [reply cmd/operator/app/operator.go]: @lonelyCZ 

This package contains a lot of reduant code for the operator, such as the `OverrideManager`, `ResourceInterpreter` and `Options` in the context.go. So the operator context is different from the `Karmada Controller Manager` or `Karmada Agent` context. 

If we want to reuse it, It seems that we need to refactor the code in the karmada main repo.

[the remark this answers is not in the document]
Comment (carlory) [reply cmd/operator/app/operator.go]: I'm not sure, I think there is a high probability that there will be multiple controller logics, such as those responsible for adding new clusters.

File: cmd/controller-manager/app/options/options.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	MetricsBindAddress string
+
+	// The number of karmada controller reconcilers that are allowed to sync concurrently.
+	ConcurrentKarmadaSyncs int
+
+	Logs *logs.Options
+}
+
+// NewControllerManagerOptions creates a new ControllerManagerOptions with a default config.
+func NewControllerManagerOptions() (*ControllerManagerOptions, error) {
```
Comment (lonelyCZ) [inline cmd/controller-manager/app/options/options.go:53]:
```suggestion
func NewControllerManagerOptions() *ControllerManagerOptions {
```

File: cmd/controller-manager/app/controllermanager.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+			}
+			return nil
+		},
+	}
+
+	fs := cmd.Flags()
+	namedFlagSets := o.Flags(KnownControllers(), ControllersDisabledByDefault.List())
+
+	// Add the flag(--kubeconfig) that is added by controller-runtime
+	namedFlagSets.FlagSet("misc").AddGoFlagSet(flag.CommandLine)
```
Comment (lonelyCZ) [inline cmd/controller-manager/app/controllermanager.go:85]: It seems unnecessary to be a single flag group.

File: cmd/controller-manager/app/controllermanager.go [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+	// never reach here
+	return nil
+}
+
+// InitFunc is used to launch a particular controller. It returns a controller
+// that can optionally implement other interfaces so that the controller manager
+// can support the requested features.
+// Any error returned will cause the controller process to `Fatal`
+// The bool indicates whether the controller was enabled.
+type InitFunc func(controllerCtx ControllerContext) (enabled bool, err error)
```
Comment (lonelyCZ) [inline cmd/controller-manager/app/controllermanager.go:144]: These functions could be moved to `core.go`, or putting `pkgcontroller/context/context.go` uniformly.

**Comment by carlory** [comment]
@lonelyCZ updated

File: cmd/controller-manager/app/options/options.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+		KubeAPIQPS:             50,
+		KubeAPIBurst:           100,
+		ConcurrentKarmadaSyncs: 5,
+		Logs:                   logs.NewOptions(),
+	}
+	return &o
+}
+
+// Flags returns flags for a specific APIServer by section name
+func (o *ControllerManagerOptions) Flags(allControllers []string, disabledByDefaultControllers []string) cliflag.NamedFlagSets {
```
Comment (lonelyCZ) [inline cmd/controller-manager/app/options/options.go:76]:
```suggestion
func (o *ControllerManagerOptions) AddFlags(allControllers []string, disabledByDefaultControllers []string) cliflag.NamedFlagSets {
```

File: cmd/controller-manager/app/options/options.go [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	fss := cliflag.NamedFlagSets{}
+
+	genericfs := fss.FlagSet("generic")
```
Comment (lonelyCZ) [inline cmd/controller-manager/app/options/options.go:79]: Don't set flagset in `AddFlags()`

File: cmd/controller-manager/app/options/options.go [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+	// MetricsBindAddress is the TCP address that the controller should bind to
+	// for serving prometheus metrics.
+	// It can be set to "0" to disable the metrics serving.
+	// Defaults to ":8080".
+	MetricsBindAddress string
+
+	// The number of karmada controller reconcilers that are allowed to sync concurrently.
+	ConcurrentKarmadaSyncs int
+
+	Logs *logs.Options
```
Comment (lonelyCZ) [inline cmd/controller-manager/app/options/options.go:49]: Why set `Logs` here.

File: cmd/controller-manager/app/core.go [hunk 8]
```diff
@@ -0,0 +1,30 @@
+package app
```
Comment (lonelyCZ) [inline cmd/controller-manager/app/core.go:1]: It is unnecessary to put these code in `core.go`. We can remove it.

File: cmd/controller-manager/app/controllermanager.go [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+			verflag.PrintAndExitIfRequested()
+
+			// Activate logging as soon as possible, after that
+			// show flags with the final logging configuration.
+			if err := logsapi.ValidateAndApply(o.Logs, utilfeature.DefaultFeatureGate); err != nil {
+				return err
+			}
+
+			// To help debugging, print flags before further processing.
+			cliflag.PrintFlags(cmd.Flags())
```
Comment (lonelyCZ) [inline cmd/controller-manager/app/controllermanager.go:60]: It is better to make it easy as far as possible. We can introduce them when we need them.

File: cmd/operator/app/options/options.go [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+	KubeAPIBurst int32
+	// ResyncPeriod is the base frequency the informers are resynced.
+	// Defaults to 0, which means the created informer will never do resyncs.
+	ResyncPeriod metav1.Duration
+	// MetricsBindAddress is the TCP address that the controller should bind to
+	// for serving prometheus metrics.
+	// It can be set to "0" to disable the metrics serving.
+	// Defaults to ":8080".
+	MetricsBindAddress string
+	// The number of karmada controller reconcilers that are allowed to sync concurrently.
```
Comment (lonelyCZ) [inline cmd/operator/app/options/options.go:43]:
```suggestion
	// ConcurrentKarmadaSyncs is the number of karmada objects that are allowed to sync concurrently.
```

File: cmd/operator/app/options/options.go [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+	fs.StringSliceVar(&o.Controllers, "controllers", o.Controllers, fmt.Sprintf(""+
+		"A list of controllers to enable. '*' enables all on-by-default controllers, 'foo' enables the controller "+
+		"named 'foo', '-foo' disables the controller named 'foo'.\nAll controllers: %s\nDisabled-by-default controllers: %s",
+		strings.Join(allControllers, ", "), strings.Join(disabledByDefaultControllers, ", ")))
+	fs.IntVar(&o.ConcurrentKarmadaSyncs, "concurrent-karmada-syncs", o.ConcurrentKarmadaSyncs, "The number of karmada controller reconcilers that are allowed to sync concurrently.")
+	options.BindLeaderElectionFlags(&o.LeaderElection, fs)
+}
+
+// Validate is used to validate the options and config before launching the controller manager
+func (o *Options) Validate(allControllers []string, disabledByDefaultControllers []string) error {
```
Comment (lonelyCZ) [inline cmd/operator/app/options/options.go:83]:
```suggestion
func (o *Options) Validate() error {
```

File: config/deploy/namespace.yaml [hunk 12]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: karmada-system
```
Comment (lonelyCZ) [inline config/deploy/namespace.yaml:4]: Add a newline.

File: config/deploy/operator.yaml [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+        app: karmada-operator
+    spec:
+      serviceAccountName: karmada-operator
+      containers:
+      - command:
+        - /bin/karmada-operator
+        - --v=4
+        image: ghcr.io/karmada-io/karmada-operator
+        imagePullPolicy: Always
+        name: karmada-operator
```
Comment (lonelyCZ) [inline config/deploy/operator.yaml:41]: Add a newline.

File: config/samples/karmada.yaml [hunk 14]
```diff
@@ -0,0 +1,5 @@
+apiVersion: operator.karmada.io/v1alpha1
+kind: Karmada
+metadata:
+  name: karmada
+  namespace: karmada-system
```
Comment (lonelyCZ) [inline config/samples/karmada.yaml:5]: Add a newline.

File: config/deploy/operator.yaml [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+    metadata:
+      labels:
+        app: karmada-operator
+    spec:
+      serviceAccountName: karmada-operator
+      containers:
+      - command:
+        - /bin/karmada-operator
+        - --v=4
+        image: ghcr.io/karmada-io/karmada-operator
```
Comment (lonelyCZ) [inline config/deploy/operator.yaml:39]: The image registry should be unified with main repository.

**Comment by carlory** [comment]
cc @lonelyCZ

**Comment by carlory** [comment]
change: import sharedcli pkg of the karmada repo.

cc @lonelyCZ

**Comment by lonelyCZ** [comment]
Hi, @carlory , could you help to list what dependencies we have to import from main repo for the operator?

**Comment by carlory** [comment]
@lonelyCZ ok! I will do it later.

**Comment by carlory** [comment]
close it due to karmada#2973

**Pull request closed without merging**